hackage-upload/README.md

```diff
@@ -0,0 +1,56 @@
+# hackage-upload
+
+GitHub Action: Hackage upload
+
+Uploads an sdist archive to Hackage.
+
+## Inputs
+
+* `archive` (required): The archive created by `stack sdist` or `cabal sdist` to upload. (globs allowed)
+
+* `token` (required): An auth token from Hackage, which can be generated at `https://hackage.haskell.org/user/$USERNAME/manage`
+
+* `candidate` (optional): Whether to upload as a package candidate or not.
+    * Defaults to `true` because Hackage uploads are permanent, and it's usually not a good idea to do irreversible actions in an automatic pipeline. But if you absolutely want to skip the candidate step, set this to `false`.
+
+## Outputs
+
+N/A
+
+## Example
+
+```yaml
+name: Release
+on:
+  push:
+    branches:
+      - 'releases/*'
+
+jobs:
+  hlint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - run: stack sdist
+
+    - uses: haskell/actions/hackage-upload@main
+      with:
+        archive: my-library-*.tar.gz
+        token: ${{ secrets.HACKAGE_TOKEN }}
```

Comment:
Would it be possible to add a url option here so that self hosted hackage instances could be used? That would be quite useful and I believe some people _do_ use those.